### PR TITLE
Use subsystems dict lookup instead of handle

### PR DESCRIPTION
**Pull request by glimchb**
Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

**Review by mestery — APPROVED** [review]
image

File: frontend.go [hunk 1]
```diff
@@ -193,7 +200,7 @@ func (s *server) DeleteNVMeController(ctx context.Context, in *pb.DeleteNVMeCont
 		return nil, fmt.Errorf("error finding controller %s", in.ControllerId.Value)
 	}
 	params := MrvlNvmSubsysRemoveCtrlrParams{
-		Subnqn: in.GetControllerId().GetValue(),
+		Subnqn: controller.Spec.SubsystemId.Value,
```
[the remark this answers is not in the document]
Comment (glimchb) [reply frontend.go:203]: fixed

File: frontend.go [hunk 2]
```diff
@@ -261,7 +274,7 @@ func (s *server) GetNVMeController(ctx context.Context, in *pb.GetNVMeController
 		return nil, fmt.Errorf("error finding controller %s", in.ControllerId.Value)
 	}
 	params := MrvlNvmGetCtrlrInfoParams{
-		Subnqn: in.GetControllerId().GetValue(),
+		Subnqn: controller.Spec.SubsystemId.Value,
```
[the remark this answers is not in the document]
Comment (glimchb) [reply frontend.go:277]: fixed

File: frontend.go [hunk 3]
```diff
@@ -386,8 +414,14 @@ func (s *server) ListNVMeNamespace(ctx context.Context, in *pb.ListNVMeNamespace
...
 	log.Printf("GetNVMeNamespace: Received from client: %v", in)
+	namespace, ok := namespaces[in.NamespaceId.Value]
+	if !ok {
+		err := fmt.Errorf("unable to find key %s", in.NamespaceId.Value)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
 	params := MrvlNvmGetNsInfoParams{
-		SubNqn: in.GetNamespaceId().GetValue(),
+		SubNqn: namespace.Spec.SubsystemId.Value,
```
[the remark this answers is not in the document]
Comment (glimchb) [reply frontend.go:424]: fixed

File: frontend.go [hunk 4]
```diff
@@ -404,8 +438,14 @@ func (s *server) GetNVMeNamespace(ctx context.Context, in *pb.GetNVMeNamespaceRe
...
 	log.Printf("NVMeNamespaceStats: Received from client: %v", in)
+	namespace, ok := namespaces[in.NamespaceId.Value]
+	if !ok {
+		err := fmt.Errorf("unable to find key %s", in.NamespaceId.Value)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
 	params := MrvlNvmGetNsStatsParams{
-		SubNqn: in.GetNamespaceId().GetValue(),
+		SubNqn: namespace.Spec.SubsystemId.Value,
```
[the remark this answers is not in the document]
Comment (glimchb) [reply frontend.go:448]: fixed

File: frontend.go [hunk 5]
```diff
@@ -279,8 +292,12 @@ func (s *server) GetNVMeController(ctx context.Context, in *pb.GetNVMeController
 
 func (s *server) NVMeControllerStats(ctx context.Context, in *pb.NVMeControllerStatsRequest) (*pb.NVMeControllerStatsResponse, error) {
 	log.Printf("NVMeControllerStats: Received from client: %v", in)
+	controller, ok := controllers[in.Id.Value]
+	if !ok {
+		return nil, fmt.Errorf("error finding controller %s", in.Id.Value)
+	}
 	params := MrvlNvmGetCtrlrStatsParams{
-		Subnqn: in.GetId().GetValue(),
+		Subnqn: controller.Spec.SubsystemId.Value,
```
[the remark this answers is not in the document]
Comment (glimchb) [reply frontend.go:300]: fixed

**Pull request merged**